File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.js
function solution(arr, n) {
    var answer = [];
    arr.map((item,index)=>{
        if( arr.length % 2 === 0 ){
            if( index % 2 === 1 ){
                answer.push(item + n)
            }else{
                answer.push(item)
            }
        }else if( arr.length % 2 === 1 ){
            if( index % 2 === 0 ){
                answer.push(item + n)
            }else{
                answer.push(item)
            }
        }
    })
    return answer;
}

File: 프로그래머스/0/181854. 배열의 길이에 따라 다른 연산하기/배열의 길이에 따라 다른 연산하기.java
class Solution {
    public int[] solution(int[] arr, int n) {
        int[] answer = new int[arr.length];
        for(int i = 0; i< arr.length; i++){
            answer[i] = arr[i];
            if(arr.length % 2 == 0 && i % 2 != 0){
                answer[i] += n;
            }
            if(arr.length % 2 != 0 && i % 2 == 0){
                answer[i] += n;
            }
        }
        return answer;
    }
}